willa/chatbot/graph_manager.py

```diff
@@ -0,0 +1,165 @@
+"""Manages the shared state and workflow for Willa chatbots."""
+
+import logging
+from typing import Optional, Annotated, NotRequired
+from typing_extensions import TypedDict
+
+from langchain_core.language_models import BaseChatModel
+from langchain_core.messages import HumanMessage, AIMessage, SystemMessage
+from langchain_core.vectorstores.base import VectorStore
+from langchain_core.runnables.config import RunnableConfig
+from langgraph.checkpoint.memory import InMemorySaver
+from langgraph.graph import START, StateGraph, add_messages
+from langgraph.graph.state import CompiledStateGraph
+from langgraph.graph.message import AnyMessage
+
+from willa.config import CONFIG, get_lance, get_ollama
+from willa.tind import format_tind_context
+
+LOGGER = logging.getLogger(__name__)
+"""The logging instance used for graph manager log messages."""
+
+STORE = get_lance()
+"""The LanceDB instance used for a vector store."""
+
+MODEL = get_ollama()
+"""The ChatOllama instance used for a chat model."""
+
+with open(CONFIG['PROMPT_TEMPLATE'], encoding='utf-8') as f:
+    _SYS_PROMPT: str = f.read()
+    """The system prompt text."""
+
+
+class WillaChatbotState(TypedDict):
+    """State for the Chatbot LangGraph workflow."""
+    messages: Annotated[list[AnyMessage], add_messages]
+    context: NotRequired[str]
+    search_query: NotRequired[str]
+    tind_metadata: NotRequired[str]
+
+
+class GraphManager:  # pylint: disable=too-few-public-methods
+    """Manages the shared LangGraph workflow for all chatbot instances."""
+
+    def __init__(self) -> None:
+        self.memory = InMemorySaver()
+        self._current_vector_store: Optional[VectorStore] = STORE
+        self._current_model: Optional[BaseChatModel] = MODEL
```

Comment:
What if we called them during the init of the GraphManager, instead of having globals?

```suggestion
        self._current_vector_store: Optional[VectorStore] = get_lance()
        self._current_model: Optional[BaseChatModel] = get_ollama()
```